src/coreclr/src/gc/unix/gcenv.unix.cpp

```diff
@@ -834,7 +826,7 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
     cacheSize = std::max(cacheSize, ( size_t) sysconf(_SC_LEVEL4_CACHE_SIZE));
 #endif
 
-#if defined(TARGET_LINUX) && !defined(HOST_ARM)
+#if defined(TARGET_LINUX) && !defined(HOST_ARM) && !defined(TARGET_OSX)
```

Comment:
```suggestion
#if defined(TARGET_LINUX) && !defined(HOST_ARM)
```

`TARGET_OSX` can never be set together with `TARGET_LINUX`